hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -408,19 +420,25 @@ public Pair<Option<String>, JavaRDD<WriteStatus>> syncOnce() throws IOException
         .build();
     String instantTime = metaClient.createNewInstantTime();
 
-    Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> srcRecordsWithCkpt = readFromSource(instantTime);
+    InputBatch inputBatch = readFromSource(instantTime);
+
+    if (inputBatch != null) {
+      final JavaRDD<HoodieRecord> recordsFromSource;
+      if (rowBulkInsert) {
```

Comment:
we might need some fixes here. 
looks like we call 
```
setupWriteClient(recordsFromSource);
```
below. 
within that, we might estimate the size and set some write configs. 
So, even for row writer, we might have to do that.

Reply:
If you feel, thats too much work/ or involves too much change, let me know. lets see if we can punt it. but lets take a stab at fixing it.

Reply:
I think it's too much for now.